Compilation fails at src/fdoscrets/objects/SessionCipher.cpp

### Have you searched for an existing issue?

- [x] Yes, I tried searching and reviewed the pinned issues

### Brief Summary

Compilation fails at src/fdosecrets/objects/SessionCipher.cpp.


### Steps to Reproduce

Try compiling.

### Expected Versus Actual Behavior

The compilation fails instead of succeding:

```
====================[ Build | keepassxc | Debug ]===============================
/snap/clion/439/bin/cmake/linux/x64/bin/cmake --build /home/adrien/projects/keepassxc/cmake-build-debug --target keepassxc -- -j 2
[2/4] Building CXX object src/fdosecrets/CMakeFiles/fdosecrets.dir/objects/SessionCipher.cpp.o
FAILED: [code=1] src/fdosecrets/CMakeFiles/fdosecrets.dir/objects/SessionCipher.cpp.o 
/usr/bin/c++ -DQT_ACCESSIBILITY_SUPPORT_LIB -DQT_CORE_LIB -DQT_DBUS_LIB -DQT_DEVICEDISCOVERY_SUPPORT_LIB -DQT_EDID_SUPPORT_LIB -DQT_EGLFSDEVICEINTEGRATION_LIB -DQT_EGL_SUPPORT_LIB -DQT_EVENTDISPATCHER_SUPPORT_LIB -DQT_FB_SUPPORT_LIB -DQT_FONTDATABASE_SUPPORT_LIB -DQT_GLX_SUPPORT_LIB -DQT_GUI_LIB -DQT_INPUT_SUPPORT_LIB -DQT_LINUXACCESSIBILITY_SUPPORT_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_DEPRECATED_WARNINGS -DQT_NO_EXCEPTIONS -DQT_PLATFORMCOMPOSITOR_SUPPORT_LIB -DQT_SERVICE_SUPPORT_LIB -DQT_STRICT_ITERATORS -DQT_SVG_LIB -DQT_THEME_SUPPORT_LIB -DQT_WIDGETS_LIB -DQT_XCB_QPA_LIB_LIB -DQT_XKBCOMMON_SUPPORT_LIB -I/home/adrien/projects/keepassxc/cmake-build-debug/src/fdosecrets/fdosecrets_autogen/include -I/home/adrien/projects/keepassxc/src -I/home/adrien/projects/keepassxc/cmake-build-debug/src -I/home/adrien/projects/keepassxc/src/fdosecrets -I/home/adrien/projects/keepassxc/cmake-build-debug/src/fdosecrets -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include -isystem /usr/include/PCSC -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/libusb-1.0 -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/qt5 -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/qt5/QtCore -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/tools/qt5/mkspecs/linux-g++ -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/qt5/QtWidgets -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/qt5/QtGui -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/qt5/QtEglFSDeviceIntegration -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/qt5/QtEventDispatcherSupport -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/qt5/QtServiceSupport -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/qt5/QtDBus -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/qt5/QtThemeSupport -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/qt5/QtFontDatabaseSupport -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/qt5/QtFbSupport -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/qt5/QtEglSupport -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/qt5/QtInputSupport -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/qt5/QtDeviceDiscoverySupport -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/qt5/QtPlatformCompositorSupport -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/qt5/QtSvg -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/qt5/QtXkbCommonSupport -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/qt5/QtLinuxAccessibilitySupport -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/qt5/QtAccessibilitySupport -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/qt5/QtEdidSupport -isystem /home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/qt5/QtGlxSupport -fno-common -fopenmp -Wall -Wextra -Wundef -Wpointer-arith -Wno-long-long -Wformat=2 -Wmissing-format-attribute -fvisibility=hidden -fvisibility-inlines-hidden -Wshadow-compatible-local -Wshadow-local -Werror -Wno-deprecated-enum-enum-conversion -Wno-error=deprecated  -fstack-protector-strong -Wnon-virtual-dtor -Wold-style-cast -Woverloaded-virtual -Werror=format-security -Wcast-align -fsized-deallocation -Wno-deprecated-declarations -g -std=gnu++20 -fPIC -fdiagnostics-color=always -fPIC -MD -MT src/fdosecrets/CMakeFiles/fdosecrets.dir/objects/SessionCipher.cpp.o -MF src/fdosecrets/CMakeFiles/fdosecrets.dir/objects/SessionCipher.cpp.o.d -o src/fdosecrets/CMakeFiles/fdosecrets.dir/objects/SessionCipher.cpp.o -c /home/adrien/projects/keepassxc/src/fdosecrets/objects/SessionCipher.cpp
/home/adrien/projects/keepassxc/src/fdosecrets/objects/SessionCipher.cpp: In constructor ?FdoSecrets::DhIetf1024Sha256Aes128CbcPkcs7::DhIetf1024Sha256Aes128CbcPkcs7(const QByteArray&)?:
/home/adrien/projects/keepassxc/src/fdosecrets/objects/SessionCipher.cpp:44:113: error: invalid use of incomplete type ?class Botan::DL_Group?
   44 |             m_privateKey.reset(new Botan::DH_PrivateKey(*randomGen()->getRng(), Botan::DL_Group("modp/ietf/1024")));
      |                                                                                                                 ^
In file included from /home/adrien/projects/keepassxc/src/fdosecrets/objects/SessionCipher.cpp:26:
/home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/botan/dh.h:17:7: note: forward declaration of ?class Botan::DL_Group?
   17 | class DL_Group;
      |       ^~~~~~~~
/home/adrien/projects/keepassxc/src/fdosecrets/objects/SessionCipher.cpp: In member function ?bool FdoSecrets::DhIetf1024Sha256Aes128CbcPkcs7::updateClientPublicKey(const QByteArray&)?:
/home/adrien/projects/keepassxc/src/fdosecrets/objects/SessionCipher.cpp:71:38: error: no matching function for call to ?Botan::PK_Ops::Key_Agreement::agree(int, const uint8_t*, int, unsigned char*, std::vector<unsigned char, Botan::secure_allocator<unsigned char> >::size_type)?
   71 |             auto aesKey = dhka->agree(16,
      |                           ~~~~~~~~~~~^~~~
   72 |                                       reinterpret_cast<const uint8_t*>(clientPublicKey.constData()),
      |                                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   73 |                                       clientPublicKey.size(),
      |                                       ~~~~~~~~~~~~~~~~~~~~~~~
   74 |                                       salt.data(),
      |                                       ~~~~~~~~~~~~
   75 |                                       salt.size());
      |                                       ~~~~~~~~~~~~
/home/adrien/projects/keepassxc/src/fdosecrets/objects/SessionCipher.cpp:71:38: note: there is 1 candidate
In file included from /home/adrien/projects/keepassxc/src/fdosecrets/objects/SessionCipher.cpp:32:
/home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/botan/pk_ops.h:141:38: note: candidate 1: ?virtual Botan::secure_vector<unsigned char> Botan::PK_Ops::Key_Agreement::agree(std::size_t, std::span<const unsigned char>, std::span<const unsigned char>)?
  141 |       virtual secure_vector<uint8_t> agree(size_t key_len,
      |                                      ^~~~~
/home/adrien/projects/keepassxc/cmake-build-debug/vcpkg_installed/x64-linux/include/botan/pk_ops.h:141:38: note: candidate expects 3 arguments, 5 provided
ninja: build stopped: subcommand failed.
```

### KeePassXC Debug Information

```Text

```

### Operating System

Linux

### Linux Desktop Environment

Gnome

### Linux Windowing System

X11

--- a/src/fdosecrets/objects/SessionCipher.cpp
+++ b/src/fdosecrets/objects/SessionCipher.cpp
@@ -25,7 +25,7 @@
 #include <QDebug>
 #include <botan/dh.h>
 
-#ifdef WITH_XC_BOTAN3
+#ifdef WITH_BOTAN3
 #include <botan/dl_group.h>
 #include <botan/pubkey.h>
 #else
@@ -58,7 +58,7 @@ namespace FdoSecrets
 
         try {
             Botan::secure_vector<uint8_t> salt(32, '\0');
-#ifdef WITH_XC_BOTAN3
+#ifdef WITH_BOTAN3
             Botan::PK_Key_Agreement dhka(*m_privateKey, *randomGen()->getRng(), "HKDF(SHA-256)", "");
             auto aesKey = dhka.derive_key(16,
                                           reinterpret_cast<const uint8_t*>(clientPublicKey.constData()),

